typescript/web/next.config.js

```diff
@@ -72,16 +71,28 @@ module.exports = withSentryConfig(
             http2: false,
             module: false,
             net: false,
-            tls: false
+            tls: false,
           },
         };
         config.plugins = [
           ...(config?.plugins ?? []),
           new NodePolyfillPlugin({
             excludeAliases: ["console"],
           }),
+          {
+            apply: (compiler) => {
+              compiler.hooks.afterEmit.tap("AfterEmitPlugin", (compilation) => {
+                exec(
+                  'yarn apollo codegen:generate --localSchemaFile=../../data/__generated__/schema.graphql --target=typescript --tagName=gql --includes="src/**/*.tsx"',
+                  (err, stdout, stderr) => {
+                    if (stdout) process.stdout.write(stdout);
+                    if (stderr) process.stderr.write(stderr);
+                  }
+                );
+              });
+            },
```

Comment:
As I understand this code, the build wouldn't fail if codegen had an error. Is that right?

Also, can you factorize it into a `generate` function which would simply call `yarn generate`.
This `generate` command would make sure that **_both_** the apollo generator and the GraphQL codegen would be called (so that we could also gitignore the output of GraphQL very soon).

Reply:
Do you mean the build resulting of the command `yarn build:web` ?

Reply:
In the sentence _`the build wouldn't fail if codegen had an error`_ I meant the process which called to this code (whichever it is, dev or build mode).

If your question objective was to make sure that I didn't meant _"please make this function called also during `yarn build:web`"_, then no.
But why not if it's possible and don't complexifies things 🚀  At-least this would be more respectful of the DRY principle...

Reply:
Then yes you are right it would not fail, it just fires an error in console but it's emitted after the build

Reply:
Why wouldn't you want it to fail?

Reply:
Because we never discuss it 🤷

Reply:
Because that's a common rule of thumb we do (_nearly_) everywhere.
In a robust process, regardless of the language its been written with, you should fail as soon as an error occur in a non-optional step. Not ignore it and print a message on stderr.
They are cases where you'll not want the error to completely stop your process (_mostly with servers and GUI_) but these will be handled with `catch` clauses placed at the highest level in the stack so that you keep the lower-levels robust and side-effect free.
For example, in our specific case, we dont wan't to kill the high-level watcher which restarts the "build" at each change but we want to prevent the rest of the "build" from continuing if we haven't generated the files.